File: problems/1863-sum-of-all-subset-xor-totals/solution.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();
        int totalSum = 0;
        for (int i = 0; i < (1 << n); ++i) {
            int currentXOR = 0;
            for (int j = 0; j < n; ++j) {
                if ((i >> j) & 1) {
                    currentXOR ^= nums[j];
                }
            }
            totalSum += currentXOR;
        }
        return totalSum;
    }
};

File: problems/1863-sum-of-all-subset-xor-totals/solution.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        n = len(nums)
        total_xor_sum = 0
        for i in range(1 << n):
            current_xor_sum = 0
            for j in range(n):
                if (i >> j) & 1:
                    current_xor_sum ^= nums[j]
            total_xor_sum += current_xor_sum
        return total_xor_sum